src/js/components/DataTable/buildState.js

```diff
@@ -173,33 +173,45 @@ export const buildFooterValues = (columns, data) => {
 
 // looks at the groupBy property of each data object and returns an
 // array with one item for each unique value of that property.
-export const buildGroups = (columns, data, groupBy) => {
+export const buildGroups = (columns, data, groupBy, primaryProperty) => {
   let result;
-  if (groupBy) {
+  if (groupBy?.property || typeof groupBy === 'string') {
     result = [];
     const groupMap = {};
     data.forEach((datum) => {
+      const key = datumValue(datum, primaryProperty);
+      const isGroup = (key && groupBy.expandable?.includes(key));
+
       const groupByProperty = groupBy.property ? groupBy.property : groupBy;
-      const groupValue = datumValue(datum, groupByProperty);
-      if (!groupMap[groupValue]) {
-        const group = { data: [], datum: {}, key: groupValue };
+      const groupValue = isGroup ? key : datumValue(datum, groupByProperty);
+      if (!groupMap[isGroup ? key : groupValue]) {
+        const group = {
+          data: [],
+          datum: isGroup ? datum : {},
+          key: groupValue,
+        };
         group.datum[groupByProperty] = groupValue;
         result.push(group);
         groupMap[groupValue] = group;
       }
-      groupMap[groupValue].data.push(datum);
+      if (!isGroup) groupMap[groupValue].data.push(datum);
     });
 
     // include any aggregate column values across the data for each group
-    columns.forEach((column) => {
-      if (column.aggregate) {
-        result.forEach((group) => {
-          const { datum } = group;
-          datum[column.property] = aggregateColumn(column, group.data);
-        });
-      }
-    });
+    if (!groupBy.expandable) {
```

Comment:
Can we include a comment about why we don't do this when `expandable`?

Reply:
done